src/index.js

```diff
@@ -49,14 +49,11 @@ class HaStore extends EventEmitter {
    * @param {object} params (Optional)The Request parameters
    * @returns {Promise} The eventual single record
    */
-  get(ids, params = {}, agg = null) {
+  async get(ids, params = {}, agg = null) {
     if (params === null) params = {};
-    const uid = Math.random().toString(36);
     const requestIds = (Array.isArray(ids)) ? ids : [ids];
-    const promises = requestIds.map((id, i) => {
-      return this.queue.push(id, params, agg, (this.config.batch === null && i === requestIds.length - 1), uid);
-    });
-    return Promise.all(promises)
+    const key = contextKey(this.config.uniqueParams, params);
+    return Promise.all(this.queue.getHandles(key, requestIds, params, agg))
```

Comment:
Ideally, this would be `allSettled`, but this means adding a polyfill until node 14 becomes LTS

Reply:
Under workds as a fast follow in `v2.1.0`